lib/screens/settings/language.dart

```diff
@@ -50,27 +52,76 @@ class LanguagePage extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final loc = AppLocalizations.of(context)!;
-    final locale = ref.watch(languageProvider);
-    final supported = List.of(AppLocalizations.supportedLocales
-        .where((l) => !kSkipLocales.contains(l)));
-    supported.sort((a, b) => (a.languageCode == 'en' ? 0 : 1)
-        .compareTo(b.languageCode == 'en' ? 0 : 1));
+    final localizations = AppLocalizations.of(context)!;
+    final appLocale =
+        ref.watch(languageProvider) ?? Localizations.localeOf(context);
+    var supportedLocales = _getSupportedLocales(appLocale);
 
     return Scaffold(
-      appBar: AppBar(title: Text(loc.settingsLanguage)),
+      appBar: AppBar(title: Text(localizations.settingsLanguage)),
       body: ListView.separated(
-        itemCount: supported.length,
+        itemCount: supportedLocales.length,
         itemBuilder: (context, index) => ListTile(
-          title: Text(kLanguageNames[supported[index]] ??
-              supported[index].toLanguageTag()),
-          trailing: supported[index] == locale ? Icon(Icons.check) : null,
+          title: Text(kLanguageNames[supportedLocales[index]] ??
+              supportedLocales[index].toLanguageTag()),
+          trailing:
+              supportedLocales[index] == appLocale ? Icon(Icons.check) : null,
           onTap: () {
-            ref.read(languageProvider.notifier).set(supported[index]);
+            ref.read(languageProvider.notifier).set(supportedLocales[index]);
           },
         ),
         separatorBuilder: (context, index) => Divider(),
       ),
     );
   }
+
+  List<Locale> _getSupportedLocales(Locale appLocale) {
```

Comment:
This all looks much better and understandable! One thing I'd ask for, please move this method above the `build()`: I prefer having everything declared before usage, so rendering methods usually come last.